File: pyg_cart_web/src/main/webapp/js/service/cartService.js
app.service('cartService',function($http){
    //查找购物车列表
    this.findcartlist=function () {
      return $http.get('../cart/findcartlist.do');
  }
    //添加删除购物车数据
    this.addGoodstoCartList=function (itemId,num) {
        return $http.get('../cart/addGoodstoCartList.do?itemId='+itemId+'&num='+num);
    }
    //获取地址列表
    this.findAddressList=function () {
        return $http.get('../address/findListByUser.do');
    }
    //生成order订单
    this.submitOrder=function (order) {
        return $http.post('../order/add.do',order);
    }
});

File: pyg_cart_interface/src/main/java/com/pyg/cart/service/CartService.java
package com.pyg.cart.service;

import entity.Cart;

import java.util.List;

public interface CartService {

    List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    List<Cart> findCartListFromRedis(String name);

    void saveCartListToRedis(String name, List<Cart> cartList_redis);

    List<Cart> mergeCartList(List<Cart> cartList_cookie, List<Cart> cartList_redis);
}
